src/tabs/Tabs.tsx

```diff
@@ -127,16 +131,21 @@ export type TabsProps = {
 export function Tabs({
   children,
   className,
+  index,
   onChange,
   orientation = 'horizontal',
   extra,
 }: TabsProps) {
   // Filter out the nulls from the children so that tabs can be mounted conditionally
   children = Children.toArray(children).filter(child => child);
   const tabs = parseTabList(children);
-  // Initialize the selected tab to the first non-hidden tab
-  const [selectedIndex, setSelectedIndex] = useState<number>(
-    tabs.findIndex(tab => !tab.hidden)
+  
+  // Initialize the selected tab to the first non-hidden tab if there is no controlled value provided
+  const defaultValue = tabs.findIndex(tab => !tab.hidden);
+  const [selectedIndex, setSelectedIndex] = useControlledState(
+    index,
+    defaultValue,
+    () => {}
```

Comment:
I think you can simplify this even further

```
export function useControlledState<T, C = T>(value: Exclude<T, undefined>, defaultValue: Exclude<T, undefined> | undefined, onChange?: (v: C, ...args: any[]) => void): [T, (value: T) => void];
```
```suggestion
    onChange
```

Reply:
This should allow you to fully remove the explicit call to onChange below

Reply:
wow, amazing